lib/msf/base/sessions/mainframe_shell.rb

```diff
@@ -0,0 +1,155 @@
+# -*- coding: binary -*-
+require 'msf/base/sessions/command_shell'
+
+module Msf
+module Sessions
+
+###
+#
+# This class provides basic interaction with a Unix Systems Service
+# command shell on a mainframe (IBM System Z) running Z/OS
+# This session is initialized with a stream that will be used
+# as the pipe for reading and writing the command shell.
+#
+#  Date:    Oct 8, 2015
+#  Author:  Bigendian Smalls
+#
+###
+class Msf::Sessions::MainframeShell < Msf::Sessions::CommandShell
+
+  #
+  # This interface supports basic interaction.
+  #
+  include Msf::Session::Basic
+
+  #
+  # This interface supports interacting with a single command shell.
+  #
+  include Msf::Session::Provider::SingleCommandShell
+
+  ##
+  #
+  # initialize as mf shell session
+  #
+  def initialize(*args)
+    self.platform = "mainframe"
+    self.arch = "zarch"
+    self.translate_1047 = true
+    super
+  end
+
+  ##
+  #
+  # Returns the session description.
+  #
+  def desc
+    "Mainframe shell"
+  end
+
+  ##
+  #
+  # override shell_read to include decode of cp1047
+  #
+  def shell_read(length=-1, timeout=1)
+    #mfimpl
+    if self.respond_to?(:ring)
+      return Rex::Text.from_ibm1047(shell_read_ring(length,timeout))
+    end
+
+    begin
+      rv = Rex::Ttext.from_ibm1047(rstream.get_once(length, timeout))
```

Comment:
A typo here. Should be Rex::Text.from_ibm1047(rstream.get_once(length, timeout))